### PR TITLE
rgw: create sync-module instance when execute radosgw-admin data sync run

### PR DESCRIPTION
**Pull request by Leeshine**
When execute radowgw-admin data sync run, create a sync-module instance to
handle data sync

Fixes: http://tracker.ceph.com/issues/22080
Signed-off-by: lvshanchun <lvshanchun@gmail.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes, in this case, it would be "rgw: ".

File: src/rgw/rgw_admin.cc [hunk 1]
```diff
@@ -2930,7 +2930,11 @@ int main(int argc, const char **argv)
   if (raw_storage_op) {
     store = RGWStoreManager::get_raw_storage(g_ceph_context);
   } else {
-    store = RGWStoreManager::get_storage(g_ceph_context, false, false, false, false, false);
+    bool run_sync_thread = false;
+    if (opt_cmd == OPT_DATA_SYNC_RUN) {
+      run_sync_thread = true;
```
Comment (oritwas) [inline src/rgw/rgw_admin.cc]: I don't like this as running sync thread has a lot of side effects.
I prefer another fix

**Comment by Leeshine** [comment]
@oritwas ping

**Comment by Leeshine** [comment]
@oritwas ping

**Comment by Leeshine** [comment]
jenkins test this please

**Comment by Leeshine** [comment]
jenkins: retest this please

**Comment by Leeshine** [comment]
jenkins test this please

**Comment by yuriw** [comment]
[wip-yuri2-testing-2017-12-18-1657](https://trello.com/c/PgmGgEIx/213-wip-yuri2-testing-2017-12-18-1657)

**Comment by Leeshine** [comment]
jenkins test this please

**Comment by cbodley** [comment]
jenkins test docs

**Pull request merged**